### ISSUE TITLE
[build_runner]: AssetNotFoundException: gtk_flutter|lib/src/config/router/app_router.gr.dart

### ISSUE THREAD

**Issue by EddyMurphyxd** [dart-lang/build#3606]
This was caused by running the `pub run build_runner build --delete-conflicting-outputs` command.
Once I created manually the app_router.gr.dart file - the next run has succeeded.

However, the original issue has yet to be solved :'(
```bash
pub run build_runner build --delete-conflicting-outputs
Deprecated. Use `dart run` instead.
Building package executable... (2.5(4.9s)
Built build_runner:build_runner.
[INFO] Generating build script completed, took 248ms
[INFO] Precompiling build script... completed, took 5.0s
[INFO] Building new asset graph...

You have hit a bug in build_runner
Please file an issue with reproduction steps at https://github.com/dart-lang/build/issues


AssetNotFoundException: gtk_flutter|lib/src/config/router/app_router.gr.dart
package:build_runner_core/src/asset/file_based.dart 135:18           _fileForOrThrow.<fn>
dart:async                                                           _rootRunUnary
package:build/src/asset/reader.dart 53:13                            AssetReader.digest
package:build_runner_core/src/asset_graph/graph.dart 171:30          AssetGraph._setLastKnownDigests.<fn>
dart:async                                                           Future.wait.<fn>
package:build_runner_core/src/asset_graph/graph.dart 170:5           AssetGraph._setLastKnownDigests
package:build_runner_core/src/asset_graph/graph.dart 74:5            AssetGraph.build
package:build_runner_core/src/generate/build_definition.dart 253:24  _Loader.prepareWorkspace.<fn>
package:build_runner_core/src/logging/logging.dart 25:18             logTimedAsync
package:build_runner_core/src/generate/build_definition.dart 251:7   _Loader.prepareWorkspace
package:build_runner_core/src/generate/build_impl.dart 114:27        BuildImpl.create
package:build_runner_core/src/generate/build_runner.dart 34:26       BuildRunner.create
package:build_runner/src/generate/build.dart 109:17                  build
package:build_runner/src/entrypoint/build.dart 35:18                 BuildCommand._run
package:args/command_runner.dart 212:13                              CommandRunner.runCommand
package:build_runner/src/entrypoint/run.dart 26:18                   run
.dart_tool/build/entrypoint/build.dart 22:16                         main
```

* Dart SDK version: **3.1.5** (stable) (Tue Oct 24 04:57:17 2023 +0000) on "macos_arm64"
* What builder(s) you are using (or writing yourself). Try to give a short summary of what they do.

* Any other packages or constraints we should know about

**Comment by jakemac53** [comment]
It looks like this is failing due to a file not existing that it thinks should exist... do you possibly have multiple build_runner processes going, or something else that is cleaning up these generated files?

**Comment by EddyMurphyxd** [comment]
Hey, Jake!
I do not recall having multiple build runners on. At least intentionally.
So many things have changed during the last 2 days that I can not tell what has solved the problem of an empty generated files.

**Comment by workfor21** [comment]
terminal-iMac % dart run build_runner build --delete-conflicting-outputs
Building package executable... (2.1s)
Built build_runner:build_runner.
[INFO] Generating build script completed, took 642ms
[WARNING] Invalidated precompiled build script due to missing asset graph.
[INFO] Precompiling build script... completed, took 1.1s
[INFO] Building new asset graph...

You have hit a bug in build_runner
Please file an issue with reproduction steps at https://github.com/dart-lang/build/issues


AssetNotFoundException: transposys|lib/core/apiServices/api_services.g.dart
package:build_runner_core/src/asset/file_based.dart 135:18           _fileForOrThrow.<fn>
dart:async                                                           _rootRunUnary
package:build_runner_core/src/asset/file_based.dart 35:13            FileBasedAssetReader.readAsBytes.<fn>
package:build/src/asset/reader.dart 53:13                            AssetReader.digest
package:build_runner_core/src/asset_graph/graph.dart 171:30          AssetGraph._setLastKnownDigests.<fn>
dart:async                                                           Future.wait.<fn>
package:build_runner_core/src/asset_graph/graph.dart 170:5           AssetGraph._setLastKnownDigests
package:build_runner_core/src/asset_graph/graph.dart 74:5            AssetGraph.build
package:build_runner_core/src/generate/build_definition.dart 253:24  _Loader.prepareWorkspace.<fn>
package:build_runner_core/src/logging/logging.dart 25:18             logTimedAsync
package:build_runner_core/src/generate/build_definition.dart 251:7   _Loader.prepareWorkspace
package:build_runner_core/src/generate/build_impl.dart 114:27        BuildImpl.create
package:build_runner_core/src/generate/build_runner.dart 34:26       BuildRunner.create
package:build_runner/src/generate/build.dart 109:17                  build
package:build_runner/src/entrypoint/build.dart 35:18                 BuildCommand._run
package:args/command_runner.dart 212:13                              CommandRunner.runCommand
package:build_runner/src/entrypoint/run.dart 26:18                   run
.dart_tool/build/entrypoint/build.dart 62:16                         main

The file it is said to be missing does not exist in the folder.
in other words the folder it is said to be missing should be generated by this command.
